server/app/query_artifact_tree_lineage.py

```diff
@@ -51,11 +52,17 @@ def topological_sort(input_data,artifact_name_id_dict):
     output_data= list(parent_dict.values()) 
     return output_data
 
-def modify_arti_name(arti_name):
-    if "metrics" in arti_name:
+def modify_arti_name(arti_name, type):
+    # artifact_name optimization based on artifact type.["Dataset","Model","Metrics"]
+    name = ""
+    if type == "Metrics" :   # Example metrics:4ebdc980-1e7c-11ef-b54c-25834a9c665c:388 -> metrics:4ebd:388
```

Comment:
Put a try catch here . There are chances of failure .. If an error put some default name .. Like "Dataset-default" or Model-Default or something like that

Reply:
Added try except block